### PR TITLE
Updates the Action Management API to reject inflow action types such as authentication.

### PR DESCRIPTION
**Pull request by malithie**
## Purpose

Resolves https://github.com/wso2/product-is/issues/21384

## Approach
This PR updates following,
- Updates action management error codes such that they don't overlap with the error codes from core action management service
- Updates the OpenAPI contract to denote that in-flow action types are not allowed

System Impact: Nothing considerable

Customer Impact: Nothing considerable

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/action/management/v1/ActionsApi.java [hunk 1]
```diff
@@ -140,59 +140,59 @@ public Response deactivateAction(@ApiParam(value = "Name of the Action Type.",re
...
-    public Response deleteAction(@ApiParam(value = "Name of the Action Type.",required=true, allowableValues="preIssueAccessToken, preUpdatePassword, preUpdateProfile, preRegistration, authentication") @PathParam("actionType") String actionType, @ApiParam(value = "Id of the Action.",required=true) @PathParam("actionId") String actionId) {
+    public Response deleteAction(@ApiParam(value = "Name of the Action Type.",required=true, allowableValues="preIssueAccessToken, preUpdatePassword, preUpdateProfile, preRegistration") @PathParam("actionType") String actionType, @ApiParam(value = "Id of the Action.",required=true) @PathParam("actionId") String actionId) {
 
         return delegate.deleteAction(actionType,  actionId );
     }
 
     @Valid
     @GET
-    @Path("/types")
+    @Path("/{actionType}/{actionId}")
```
Comment (ashanthamara) [inline components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/action/management/v1/ActionsApi.java:150]: Shall we revert the unnecessary changes in below few lines

Comment (malithie) [reply components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/action/management/v1/ActionsApi.java:150]: but it's a generated code. what would happen if some one else generates this ? will the generator would still change the order or will it not generate that ?

Comment (ashanthamara) [reply components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/action/management/v1/ActionsApi.java:150]: If it is auto generated then we should keep this.

**Review by ashanthamara — DISMISSED** [review]

**Review by ashanthamara — APPROVED** [review]

**Review by ashanthamara — DISMISSED** [review]

**Review by ashanthamara — APPROVED** [review]
